src/utp_stream.cpp

```diff
@@ -3408,6 +3420,22 @@ bool utp_socket_impl::incoming_packet(boost::uint8_t const* buf, int size
 	return true;
 }
 
+boost::int32_t utp_socket_impl::adjust_receive_buf_size(boost::uint32_t const& min_rtt)
+{
+	// maximum value for RTT we would still scale for
+	const boost::int32_t rtt_ceil = 800000;
```

Comment:
the comment makes it sound like the effective RTT to be used in this function is ``min(min_rtt, rtt_ceil)``, but that's not the case. ``rtt_ceil`` is just used to calculate the coefficient. I suppose the clamping at the end makes any ``min_rtt`` that's higher still not scale the buffer. There's a potential robustness issue though, if ``min_rtt`` is large enough the multiplication with ``k`` might overflow.

Reply:
moved to free function, fixed clamping and types